oauth-client/src/main/java/io/streamnative/pulsar/handlers/kop/security/oauth/ClientConfig.java

```diff
@@ -58,8 +76,43 @@ public ClientConfig(Map<String, String> configs) {
             throw new IllegalArgumentException(String.format(
                     "invalid %s \"%s\": %s", OAUTH_CREDENTIALS_URL, credentialsUrlString, e.getMessage()));
         }
-
+        try {
+            this.clientInfo = loadPrivateKey();
+        } catch (IOException e) {
+            throw new IllegalArgumentException(String.format(
+                    "failed to load client credentials from %s: %s", credentialsUrlString, e.getMessage()));
+        }
         this.audience = configs.getOrDefault(OAUTH_AUDIENCE, null);
         this.scope = configs.getOrDefault(OAUTH_SCOPE, null);
     }
+
+    @VisibleForTesting
+    ClientInfo loadPrivateKey() throws IOException {
+        final URLConnection connection = getCredentialsUrl().openConnection();
+        try (InputStream inputStream = connection.getInputStream()) {
+            return CLIENT_INFO_READER.readValue(inputStream);
+        }
+    }
+
+    @Getter
+    @ToString
+    @EqualsAndHashCode
+    @NoArgsConstructor
+    @AllArgsConstructor
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public static class ClientInfo {
```

Comment:
I think it would be better to move `ClientInfo` out of `ClientConfig` because `ClientInfo` is also used in `ClientCredentialsFlow`.